src/rgw/rgw_multi_del.h

```diff
@@ -7,14 +7,29 @@
 #include "rgw_xml.h"
 #include "rgw_common.h"
 
+class RGWMultiDelObject : public XMLObj
+{
+  std::string key;
+  std::string version_id;
+  ceph::real_time last_mod_time;
+public:
+  RGWMultiDelObject() {}
+  ~RGWMultiDelObject() override {}
+  bool xml_end(const char *el) override;
+
+  const std::string& get_key() const { return key; }
+  const std::string& get_version_id() const { return version_id; }
+  const ceph::real_time& get_last_mod_time() const { return last_mod_time; }
+};
+
 class RGWMultiDelDelete : public XMLObj
 {
 public:
   RGWMultiDelDelete() :quiet(false) {}
   ~RGWMultiDelDelete() override {}
   bool xml_end(const char *el) override;
 
-  std::vector<rgw_obj_key> objects;
+  std::vector<RGWMultiDelObject> objects;
```

Comment:
ok, cool. if you're going this route, you don't necessarily need this vector. each XMLObj already tracks its children and provides a XMLObjIter to loop over them:
```c++
XMLObjIter iter = multi_delete->find_first();
for (XMLObj* obj = iter.get_next(); obj; obj = iter.get_next()) {
  auto del = static_cast<RGWMultiDelObject*>(obj);
```

Reply:
up to you